extensions/default/src/DicomWebDataSource/utils/clientManager.ts

```diff
@@ -0,0 +1,249 @@
+import { api } from 'dicomweb-client';
+import StaticWadoClient from './StaticWadoClient';
+import dcm4cheeReject from '../dcm4cheeReject';
+import { errorHandler, utils } from '@ohif/core';
+
+/**
+ * This object plays the central role in OHIF's multiple server handling ability.
+ * It stores all servers configurations and handles all request headers generations
+ */
+interface ConfigToAdd {
+  url: string;
+  qidoRoot: string;
+  wadoRoot: string;
+  staticWado: boolean;
+  singlepart: boolean;
+  name: string;
+  qidoConfig: {
+    url: string;
+    staticWado: boolean;
+    singlepart: boolean;
+    headers: object;
+  };
+  wadoConfig: {
+    url: string;
+    staticWado: boolean;
+    singlepart: boolean;
+    headers: object;
+  };
+  qidoDicomWebClient: api.DICOMwebClient | StaticWadoClient;
+  wadoDicomWebClient: api.DICOMwebClient | StaticWadoClient;
+}
+export default class ClientManager {
+  clients;
+  userAuthenticationService;
+
+  constructor({ params, query, dicomWebConfig, userAuthenticationService }) {
+    this.clients = [];
+    this.userAuthenticationService = userAuthenticationService;
+    const configArray = Array.isArray(dicomWebConfig) ? dicomWebConfig : [dicomWebConfig];
+    configArray.forEach(config => this.addConfiguration(params, query, config));
+  }
+
+  /**
+   * Sets authorization headers for all clients before queries
+   * @returns {void}
+   */
+  public setQidoHeaders(): void {
+    this.clients.forEach(
+      client => (client.qidoDicomWebClient.headers = this.getAuthorizationHeader())
+    );
+  }
+
+  /**
+   * Sets wado headers for all clients before queries
+   * @returns {void}
+   */
+  public setWadoHeaders(): void {
+    this.clients.forEach(
+      client => (client.wadoDicomWebClient.headers = this.getWadoHeader(client))
+    );
+  }
+
+  /**
+   * Sets authorization headers for wado clients before queries
+   * @returns {void}
+   */
+  public setAuthorizationHeadersForWADO(): void {
+    this.clients.forEach(
+      client => (client.wadoDicomWebClient.headers = this.getAuthorizationHeader())
+    );
+  }
+
+  /**
+   * Returns a boolean indicating if a client have reject abilities
+   * @returns {boolean} client reject support
+   */
+  public clientCanReject(name) {
+    return this.getClient(name)?.supportsReject;
+  }
+
+  /**
+   * Returns the reject function of a client
+   * @param name
+   * @returns {object} client reject object
+   */
+  public getClientRejectObject(name) {
+    const client = this.getClient(name);
+    if (client?.supportsReject) {
+      return dcm4cheeReject(client.wadoRoot);
```

Comment:
This is wrong, you should not have any pacs name in this file. Just have addClientRejectMethod and store it in the rejectObj of the client and for dcm4cheewe add it outside of this and here you return client.rejectMethod